nlohmann::json_abi_v3_11_3::detail::parse_error

Executed following commands:
`> sudo apt install libglfw3-dev libfreetype-dev nlohmann-json3-dev cmake`
```
> git clone https://github.com/h-valdes/kurve.git
Cloning into 'kurve'...
remote: Enumerating objects: 2879, done.
remote: Counting objects: 100% (103/103), done.
remote: Compressing objects: 100% (72/72), done.
remote: Total 2879 (delta 71), reused 59 (delta 31), pack-reused 2776 (from 1)
Receiving objects: 100% (2879/2879), 9.54 MiB | 29.59 MiB/s, done.
Resolving deltas: 100% (1880/1880), done.
```
`> cd kurve`
`> mkdir Release`
`> cd Release/`
```
> cmake -DCMAKE_BUILD_TYPE=Release ..
-- Found nlohmann_json: /usr/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.11.3", minimum required is "3.2.0") 
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libOpenGL.so   
** Release mode **
STATIC_FILES = share/kurve/static
-- Configuring done (0.1s)
-- Generating done (0.0s)
-- Build files have been written to: /home/dichler/bin/kurve/Release
```
```
> make
[  4%] Building CXX object CMakeFiles/kurve.dir/src/ai_model.cpp.o
[  8%] Building CXX object CMakeFiles/kurve.dir/src/arrow_mesh.cpp.o
[ 12%] Building CXX object CMakeFiles/kurve.dir/src/arrow_model.cpp.o
[ 16%] Building CXX object CMakeFiles/kurve.dir/src/border_mesh.cpp.o
[ 20%] Building CXX object CMakeFiles/kurve.dir/src/border_model.cpp.o
[ 24%] Building CXX object CMakeFiles/kurve.dir/src/display.cpp.o
[ 28%] Building CXX object CMakeFiles/kurve.dir/src/font.cpp.o
[ 32%] Building CXX object CMakeFiles/kurve.dir/src/game.cpp.o
[ 36%] Building CXX object CMakeFiles/kurve.dir/src/game_over.cpp.o
[ 40%] Building CXX object CMakeFiles/kurve.dir/src/gl_calls.cpp.o
[ 44%] Building C object CMakeFiles/kurve.dir/src/glad.c.o
[ 48%] Building CXX object CMakeFiles/kurve.dir/src/grid.cpp.o
[ 52%] Building CXX object CMakeFiles/kurve.dir/src/line_mesh.cpp.o
[ 56%] Building CXX object CMakeFiles/kurve.dir/src/line_model.cpp.o
[ 60%] Building CXX object CMakeFiles/kurve.dir/src/main.cpp.o
[ 64%] Building CXX object CMakeFiles/kurve.dir/src/menu.cpp.o
[ 68%] Building CXX object CMakeFiles/kurve.dir/src/player_manager.cpp.o
/home/dichler/bin/kurve/src/player_manager.cpp: In member function ‘GameState PlayerManager::check_score()’:
/home/dichler/bin/kurve/src/player_manager.cpp:230:21: warning: ‘second_place’ may be used uninitialized [-Wmaybe-uninitialized]
  230 |     if (first_place >= second_place + 2 && first_place >= this->max_score) {
      |         ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
/home/dichler/bin/kurve/src/player_manager.cpp:218:22: note: ‘second_place’ was declared here
  218 |     int first_place, second_place;
      |                      ^~~~~~~~~~~~
/home/dichler/bin/kurve/src/player_manager.cpp:230:5: warning: ‘first_place’ may be used uninitialized [-Wmaybe-uninitialized]
  230 |     if (first_place >= second_place + 2 && first_place >= this->max_score) {
      |     ^~
/home/dichler/bin/kurve/src/player_manager.cpp:218:9: note: ‘first_place’ was declared here
  218 |     int first_place, second_place;
      |         ^~~~~~~~~~~
[ 72%] Building CXX object CMakeFiles/kurve.dir/src/player_mesh.cpp.o
[ 76%] Building CXX object CMakeFiles/kurve.dir/src/player_model.cpp.o
[ 80%] Building CXX object CMakeFiles/kurve.dir/src/point.cpp.o
[ 84%] Building CXX object CMakeFiles/kurve.dir/src/shader.cpp.o
[ 88%] Building CXX object CMakeFiles/kurve.dir/src/side_panel.cpp.o
[ 92%] Building CXX object CMakeFiles/kurve.dir/src/stb_image.cpp.o
In file included from /home/dichler/bin/kurve/src/stb_image.cpp:2:
/home/dichler/bin/kurve/include/stb_image.h: In function ‘int stbi__zhuffman_decode_slowpath(stbi__zbuf*, stbi__zhuffman*)’:
/home/dichler/bin/kurve/include/stb_image.h:4123:10: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
 4123 |    if (b >= sizeof (z->size)) return -1; // some data was corrupt somewhere!
      |        ~~^~~~~~~~~~~~~~~~~~~
/home/dichler/bin/kurve/include/stb_image.h: In function ‘void* stbi__load_gif_main(stbi__context*, int**, int*, int*, int*, int*, int)’:
/home/dichler/bin/kurve/include/stb_image.h:6778:11: warning: variable ‘out_size’ set but not used [-Wunused-but-set-variable]
 6778 |       int out_size = 0;
      |           ^~~~~~~~
/home/dichler/bin/kurve/include/stb_image.h:6779:11: warning: variable ‘delays_size’ set but not used [-Wunused-but-set-variable]
 6779 |       int delays_size = 0;
      |           ^~~~~~~~~~~
[ 96%] Building CXX object CMakeFiles/kurve.dir/src/user.cpp.o
[100%] Linking CXX executable kurve
[100%] Built target kurve
```
Which results in this Error when exeuting `./kurve`
```
> ./kurve 
terminate called after throwing an instance of 'nlohmann::json_abi_v3_11_3::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 1, column 1: attempting to parse an empty input; check that your input string or stream contains the expected JSON
Aborted (core dumped)
```

My system:
```
> neofetch
            .-/+oossssoo+/-.               ***@*** 
        `:+ssssssssssssssssss+:`           ------------------------------ 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 24.04.2 LTS x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: Latitude 5420 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 6.8.0-57-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 1 day, 1 hour, 47 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 3510 (dpkg), 16 (snap) 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 5.2.21 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 2560x1440 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: sway 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   Terminal: terminator 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   CPU: 11th Gen Intel i7-1185G7 (8) @ 4.800GHz 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   GPU: Intel TigerLake-LP GT2 [Iris Xe Graphics] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Memory: 21578MiB / 31479MiB 
  +sssssssssdmydMMMMMMMMddddyssssssss+
   /ssssssssssshdmNNNNmyNMMMMhssssss/                              
    .ossssssssssssssssssdMMMNysssso.                               
      -+sssssssssssssssssyyyssss+-
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.

```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,26 +1,30 @@
 # Specify the minimum version for CMake
-
 cmake_minimum_required(VERSION 3.11)
+project(kurve)
+
+# Set C++ standard and compilation flags
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wno-unused-parameter -g")
 set(CMAKE_LINKER_FLAGS_DEBUG " -fno-omit-frame-pointer -fsanitize=address")
 set(CMAKE_CXX_FLAGS_RELEASE "-O3")
 
-project(kurve)
-
-# Set project version
+# Project version
 set(VERSION_MAJOR 0)
 set(VERSION_MINOR 2)
 
+# Dependencies
 find_package(Freetype REQUIRED)
 find_package(nlohmann_json 3.2.0 REQUIRED)
 find_package(OpenGL REQUIRED)
 
+# Include directories
 include_directories(
     ${PROJECT_SOURCE_DIR}/include
     ${PROJECT_SOURCE_DIR}/src
 )
+
+# Source files
 file(GLOB all_SRCS
     "${PROJECT_SOURCE_DIR}/include/*.h"
     "${PROJECT_SOURCE_DIR}/include/*.hpp"
@@ -29,14 +33,14 @@ file(GLOB all_SRCS
     ${FREETYPE_INCLUDE_DIRS}
 )
 
-# Configure the shader files
+# Shader files
 file(GLOB SHADERS
     "${CMAKE_SOURCE_DIR}/static/shader/*.vs"
     "${CMAKE_SOURCE_DIR}/static/shader/*.fs"
 )
 file(COPY ${SHADERS} DESTINATION "${CMAKE_BINARY_DIR}/shader")
 
-# Configure the font files
+# Font files
 file(GLOB FONTS
     "${CMAKE_SOURCE_DIR}/static/fonts/UbuntuMono-R.ttf"
     "${CMAKE_SOURCE_DIR}/static/fonts/UbuntuMono-B.ttf"
@@ -45,18 +49,17 @@ file(GLOB FONTS
 )
 file(COPY ${FONTS} DESTINATION "${CMAKE_BINARY_DIR}/fonts")
 
-# Configure the json settings files
+# Settings files
 file(GLOB SETTINGS
     "${CMAKE_SOURCE_DIR}/static/settings/players.json"
 )
 file(COPY ${SETTINGS} DESTINATION "${CMAKE_BINARY_DIR}/settings")
 
-set (PROJECT_SOURCES main.cpp)
-
+# Executable
 add_executable(kurve ${all_SRCS})
 
+# Link libraries
 link_libraries("-lglfw -ldl -lm")
-
 target_link_libraries(
     kurve PRIVATE 
     m 
@@ -67,54 +70,73 @@ target_link_libraries(
 )
 target_include_directories(kurve PRIVATE ${FREETYPE_INCLUDE_DIRS})
 
+# Target properties
 set_target_properties(kurve PROPERTIES
     CXX_STANDARD 17
     CXX_STANDARD_REQUIRED ON
     CXX_EXTENSIONS ON
 )
 
+# Output locations
 set(tool_dest "bin")
 set(files_dest "share/kurve/static")
 
+# Allow manual override of STATIC_FILES_PATH from command line
+if (NOT DEFINED STATIC_FILES_PATH)
+    if (CMAKE_BUILD_TYPE STREQUAL "Release" AND NOT CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+        set(STATIC_FILES_PATH "${CMAKE_INSTALL_PREFIX}/${files_dest}")
+    else()
+        set(STATIC_FILES_PATH "${CMAKE_BINARY_DIR}")
+    endif()
+endif()
+
+target_compile_definitions(kurve PRIVATE STATIC_FILES="${STATIC_FILES_PATH}")
+message("STATIC_FILES = ${STATIC_FILES_PATH}")
+
+# Debug mode settings
 if (CMAKE_BUILD_TYPE STREQUAL "Debug")
     message("** Debug mode **")
-    target_compile_definitions(kurve PRIVATE STATIC_FILES="${CMAKE_BINARY_DIR}")
     add_compile_definitions(DEBUG)
-    message("STATIC_FILES = ${CMAKE_BINARY_DIR}")
-endif (CMAKE_BUILD_TYPE STREQUAL "Debug")
+endif()
 
+# Release mode settings
 if (CMAKE_BUILD_TYPE STREQUAL "Release")
     message("** Release mode **")
-    target_compile_definitions(kurve PRIVATE STATIC_FILES="${CMAKE_INSTALL_PREFIX}/${files_dest}")
-    message("STATIC_FILES = ${files_dest}")
-    file(READ "template.desktop" filedata)
-    file(WRITE  "kurve.desktop" "${filedata}Exec=${CMAKE_INSTALL_PREFIX}/${tool_dest}/kurve\n")
-    file(READ "kurve.desktop" filedata)
-    file(WRITE  "kurve.desktop" "${filedata}Icon=${CMAKE_INSTALL_PREFIX}/${files_dest}/kurve.png\n")
-endif (CMAKE_BUILD_TYPE STREQUAL "Release")
 
+    # Generate .desktop file only if we are installing
+    if (NOT CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+        file(READ "template.desktop" filedata)
+        file(WRITE  "kurve.desktop" "${filedata}Exec=${CMAKE_INSTALL_PREFIX}/${tool_dest}/kurve\n")
+        file(READ "kurve.desktop" filedata)
+        file(WRITE  "kurve.desktop" "${filedata}Icon=${CMAKE_INSTALL_PREFIX}/${files_dest}/kurve.png\n")
+    endif()
+endif()
+
+# Snapcraft-specific path handling
 if (CMAKE_BUILD_TYPE STREQUAL "SNAPCRAFT")
     message("** SNAPCRAFT mode **")
-    target_compile_definitions(kurve PRIVATE STATIC_FILES="/snap/kurve/current/${files_dest}")
-    message("STATIC_FILES = ${files_dest}")
-endif (CMAKE_BUILD_TYPE STREQUAL "SNAPCRAFT")
+    set(STATIC_FILES_PATH "/snap/kurve/current/${files_dest}")
+    target_compile_definitions(kurve PRIVATE STATIC_FILES="${STATIC_FILES_PATH}")
+    message("STATIC_FILES = ${STATIC_FILES_PATH}")
+endif()
 
+# Install rules
 install(TARGETS kurve DESTINATION "${tool_dest}")
-
 install(FILES ${SHADERS} DESTINATION "${files_dest}/shader")
 install(FILES ${FONTS} DESTINATION "${files_dest}/fonts")
 install(FILES ${SETTINGS} DESTINATION "${files_dest}/settings")
 install(FILES "kurve.png" DESTINATION "${files_dest}")
 
-if (NOT CMAKE_BUILD_TYPE STREQUAL "SNAPCRAFT")
+# Install .desktop file if not SNAPCRAFT
+if (NOT CMAKE_BUILD_TYPE STREQUAL "SNAPCRAFT" AND NOT CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
     install(
         FILES kurve.desktop
         PERMISSIONS WORLD_EXECUTE WORLD_WRITE WORLD_READ
         DESTINATION ~/.local/share/applications
     )
-endif (NOT CMAKE_BUILD_TYPE STREQUAL "SNAPCRAFT")
+endif()
 
-## DEB Package generator
+# DEB packaging
 set(CPACK_GENERATOR "DEB")
 set(CPACK_PACKAGING_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
 set(CPACK_PACKAGE_DESCRIPTION "Remake of 'Achtung, die Kurve!'")
@@ -124,4 +146,4 @@ set(CPACK_PACKAGE_VENDOR "Hernán Valdés")
 set(CPACK_PACKAGE_CONTACT "hfvaldesg@gmail.com")
 set(CPACK_DEBIAN_PACKAGE_DEPENDS "libglfw3-dev,libfreetype-dev,nlohmann-json3-dev")
 set(CPACK_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}-${VERSION_MAJOR}.${VERSION_MINOR}")
-include (CPack)
+include(CPack)

